content/recipes/swagger.md

```diff
@@ -126,6 +126,138 @@ In order to explicitly set the type of the property, use the `type` key:
 age: number;
 ```
 
+#### Arrays
+
+When the property is an array, we must manually indicate the array type as shown below:
+
+```typescript
+@ApiProperty({ type: [String] })
+names: string[];
+```
+
+> info **Hint** Consider using the Swagger plugin (see [Plugin](/recipes/swagger#plugin) section) which will automatically detect arrays.
+
+Either include the type as the first element of an array (as shown above) or set the `isArray` property to `true`.
+
+<app-banner-enterprise></app-banner-enterprise>
+
+#### Circular dependencies
+
+When you have circular dependencies between classes, use a lazy function to provide the `SwaggerModule` with type information:
+
+```typescript
+@ApiProperty({ type: () => Node })
+node: Node;
+```
+
+> info **Hint** Consider using the Swagger plugin (see [Plugin](/recipes/swagger#plugin) section) which will automatically detect circular dependencies.
+
+#### Generics and interfaces
+
+Since TypeScript does not store metadata about generics or interfaces, when you use them in your DTOs, `SwaggerModule` may not be able to properly generate model definitions at runtime. For instance, below code won't be correctly inspected by the Swagger module:
```

Comment:
```suggestion
Since TypeScript does not store metadata about generics or interfaces, when you use them in your DTOs, `SwaggerModule` may not be able to properly generate model definitions at runtime. For instance, the following code won't be correctly inspected by the Swagger module:
```